editor/src/clj/editor/dialogs.clj

```diff
@@ -787,21 +787,37 @@
 
 (defn- sanitize-common [name]
   (-> name
-      string/trim
       (string/replace #"[/\\]" "") ; strip path separators
       (string/replace #"[\"']" "") ; strip quotes
-      (string/replace #"[<>:|?*]" ""))) ; Additional Windows forbidden characters
+      (string/replace #"[<>:|?*]" "") ; Additional Windows forbidden characters
+      string/trim))
 
 (defn sanitize-file-name [extension name]
   (let [name (sanitize-common name)]
     (cond-> name
-            ; disallow dots when extension is expected
-            (seq extension) (string/replace #"\." "")
```

Comment:
We actually want to allow dots in file names when renaming files, so that files named like `type-definition.d.tl` can be renamed without losing their meaningful dots.